pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -66,6 +67,14 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolapisv1beta1.SchemeGroupVersion.Version] = flowControlStorage
 	}
 
+	if apiResourceConfigSource.VersionEnabled(flowcontrolapisv1beta2.SchemeGroupVersion) {
+		flowControlStorage, err := p.storage(apiResourceConfigSource, restOptionsGetter)
+		if err != nil {
+			return genericapiserver.APIGroupInfo{}, false, err
+		}
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolapisv1beta2.SchemeGroupVersion.Version] = flowControlStorage
```

Comment:
I think this line is wrong (line 59 might be too). The map is from exposed version to storage object. This should be using the `flowControlStorage` computed by the v1beta1 clause.

The consequence of writing v1beta2 objects to disk is that after a rollback, apiserver wouldn't be able to read them, which is bad.

This is the mistake that storage version hash is supposed to catch-- unintentionally changing the storage version. We'd change the storage version in the subsequent release.

Reply:
I'm not seeing the bug here... storage version is determined by preferred serialization version in scheme.SetVersionPriority

You're right, the etcd data and storage version hash test would catch an incorrect storage version change here

Reply:
Looks like I finally confused myself over that nonsensical "storage" name...